lib/src/editor.dart

```diff
@@ -0,0 +1,499 @@
+  /// [ArgumentError] will be thrown.
+  @override
+  void appendToList(Iterable<Object> listPath, Object value) {
+    var yamlList = _traverseToList(listPath);
+
+    insertIntoList(listPath, yamlList.length, value);
+  }
+
+  /// Prepends [value] into the list at [listPath].
+  ///
+  /// If the element at the given path is not a [YamlList] or if the path is invalid, an
+  /// [ArgumentError] will be thrown.
+  @override
+  void prependToList(Iterable<Object> listPath, Object value) =>
+      insertIntoList(listPath, 0, value);
+
+  /// Inserts [value] into the list at [listPath], only if the element at the given path
+  /// is a list.
+  ///
+  /// [index] must be non-negative and no greater than the list's length. If the element at
+  /// the given path is not a [YamlList] or if the path is invalid, an [ArgumentError] will
+  /// be thrown.
+  @override
+  void insertIntoList(Iterable<Object> listPath, int index, Object value) {
+    var list = _traverseToList(listPath);
+    final edit = insertInList(_yaml, list, index, value);
+
+    final expectedList = updatedYamlList(
+        list, (nodes) => nodes.insert(index, yamlNodeFrom(value)));
+
+    _performEdit(edit, listPath, expectedList);
+  }
+
+  /// Changes the contents of the list at [listPath] by removing [deleteCount] items at [index], and
+  /// inserts [values] in-place.
+  ///
+  /// [index] must be non-negative and no greater than the list's length. If the element at
+  /// the given path is not a [YamlList], if the path is invalid, or if [index] + [deleteCount]
+  /// is greater than the list length, an [ArgumentError] will be thrown.
+  ///
+  /// ```dart
+  /// final doc = YamlEditor('[Jan, March, April, June]');
+  /// doc.spliceList([], 1, 0, ['Feb']); // [Jan, Feb, March, April, June]
+  /// doc.spliceList([], 4, 1, ['May']); // [Jan, Feb, March, April, May]
+  /// ```
+  @override
+  Iterable<YamlNode> spliceList(Iterable<Object> listPath, int index,
+      int deleteCount, Iterable<Object> values) {
+    var list = _traverseToList(listPath);
+
+    final nodesToRemove = list.nodes.getRange(index, index + deleteCount);
+
+    /// Perform addition of elements before removal to avoid scenarioes where
+    /// a block list gets emptied out to {} to avoid changing collection styles
+    /// where possible.
+
+    /// Reverse [values] and insert them.
+    final reversedValues = values.toList().reversed;
+    for (var value in reversedValues) {
+      insertIntoList(listPath, index, value);
+    }
+
+    for (var i = 0; i < deleteCount; i++) {
+      remove([...listPath, index + values.length]);
+    }
+
+    return nodesToRemove;
+  }
+
+  /// Removes the node at [path].
+  ///
+  /// Throws [ArgumentError] if [path] is invalid.
+  @override
+  YamlNode remove(Iterable<Object> path) {
+    var edit;
+    var expectedNode;
+    var nodeToRemove = parseAt(path);
+
+    if (path.isEmpty) {
+      expectedNode = null;
+      edit = SourceEdit(0, _yaml.length, '');
```

Comment:
Does parsing an empty do yaml document we get `null`?

Reply:
yeap!

Reply:
Maybe a one-line inline comment about this would be good :)